index.html

```diff
@@ -1177,119 +1177,294 @@ <h2 id="relationship-to-permissions-policy">
-          argument=] [=error=].
-          </li>
-          <li>Let |rootDesc| be |parameters|.{{PermissionSetParameters/descriptor}}.
-          </li>
-          <li>Let |typedDescriptor| be the object |rootDesc| refers to, <a>converted to an IDL
-          value</a> of <code>|rootDesc|.{{PermissionDescriptor/name}}</code>'s [=powerful
-          feature/permission descriptor type=]. If this throws an exception, return a [=invalid
-          argument=] [=error=].
-          </li>
-          <li>If |parameters|.{{PermissionSetParameters/state}} is an inappropriate <a>permission
-          state</a> for any implementation-defined reason, return a [=invalid argument=] [=error=].
-            <p class="note">
-              For example, <a>user agents</a> that define the "midi" <a>powerful feature</a> as
-              "always on" may choose to reject a command to set the <a>permission state</a> to
-              {{PermissionState/"denied"}} at this step.
+        <section>
+          <h4 id="extension-module-permissions">
+            The permissions Module
+          </h4>
+          <p>
+            The <dfn>permissions</dfn> module contains commands for
+            managing the remote end browser permissions.
+          </p>
+          <section>
+            <h5 id="extension-module-permissions-definition">
+              Definition
+            </h5>
+            <p>
+              [=remote end definition=]
             </p>
-          </li>
-          <li>Let |settings| be the [=current settings object=].
-          </li>
-          <li>Let |targets| be a <a>list</a> containing all <a>environment settings objects</a>
-          whose [=environment settings object/origin=] is the <a data-lt="same origin">same</a> as
-          the [=environment settings object/origin=] of |settings|.
-          </li>
-          <li>Let |tasks| be an empty <a>list</a>.
-          </li>
-          <li>For each <a>environment settings object</a> |target| in |targets|:
-            <ol>
-              <li>
-                <a>Queue a task</a> |task| on the [=permissions task source=] of |target|'s
-                [=relevant settings object=]'s [=environment settings object/global object=]'s
-                [=Window/browsing context=] to perform the following step:
+            <pre class="cddl remote-cddl">
+              PermissionsCommand = (
+                permissions.setPermission
+              )
+            </pre>
+          </section>
+          <section>
+            <h5 id="extension-module-permissions-types">
+              Types
+            </h5>
+            <section>
+              <h6 id="extension-module-type-permissions-PermissionDescriptor">
+                The permissions.PermissionDescriptor Type
+              </h6>
+              <pre class="cddl local-cddl">
+                permissions.PermissionDescriptor = {
+                  type: "permissionDescriptor",
```

Comment:
Is this needed? Can I remove it?

Reply:
Can we make it `permissions.PermissionDescriptor = text`?

Reply:
It would be problematic if we wanted to extend it in the future, e.g. with the fields of https://chromedevtools.github.io/devtools-protocol/tot/Browser/#type-PermissionDescriptor

Reply:
Ack, what did you want to remove? the type? I think it can be removed.

Reply:
Yes, just the type. OK, I'll remove it then.

Reply:
Done.